service/src/main/java/bio/terra/cbas/controllers/ControllerUtils.java

```diff
@@ -0,0 +1,43 @@
+package bio.terra.cbas.controllers;
+
+import bio.terra.cbas.dependencies.sam.SamService;
+import bio.terra.cbas.util.BearerTokenFilter;
+import java.util.Optional;
+import javax.servlet.ServletRequest;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ControllerUtils {
+  private static final Logger logger = LoggerFactory.getLogger(ControllerUtils.class);
+  private final ServletRequest servletRequest;
+  private final SamService samService;
+
+  @Autowired
+  ControllerUtils(ServletRequest servletRequest, SamService samService) {
```

Comment:
OOI, what is a ServletRequest and how are we auto-wiring it?

I believe `@Component`s are singletons throughout the system, and not recycled. Will that be a problem if more than one user is making requests and the `ServletRequest` is generated once and then shared?

Reply:
https://github.com/DataBiosphere/terra-workspace-data-service/blob/b479672d36545d0b45766b3e0e09564d077a19b8/service/src/main/java/org/databiosphere/workspacedataservice/activitylog/ActivityEventBuilder.java#L47-L48

Using WDS as reference, I removed the autowired request in favor for RequestContextAttributes, which seems to be scoped to each request (and because we're not storing anything hopefully less vulnerable to race conditions)